File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/2231_분해합/2231.py
N = int(input())

def generate(seed):
    num = seed
    while seed > 0:
        num += seed % 10
        seed //= 10
    return num

answer = 0
for i in range(N + 1):
    if generate(i) == N:
        answer = i
        break

print(answer)

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/2231_분해합/2231.cpp
#include <stdio.h>

int N;

int generate(int seed)
{
    int num = seed;
    while (seed > 0)
    {
        num += seed % 10;
        seed /= 10;
    }
    return num;
}

int main()
{
    scanf("%d", &N);

    int ans = 0;
    for (int i = 0; i <= N; i++)
    {
        if (generate(i) == N)
        {
            ans = i;
            break;
        }
    }

    printf("%d\n", ans);
    return 0;
}
